partner_communication_switzerland/wizards/portal_wizard.py

```diff
@@ -9,25 +9,73 @@
 #
 ##############################################################################
 
-from odoo import api, models
+import logging
+
+from odoo import api, fields, models
+from odoo.tools.translate import _
+from odoo.exceptions import UserError
+
+
+_logger = logging.getLogger(__name__)
 
 
 class PortalWizard(models.TransientModel):
     _inherit = 'portal.wizard'
 
+    invitation_config_id = fields.Many2one('partner.communication.config')
+
+    @api.onchange('portal_id')
+    def onchange_portal_id(self):
+        # set the values of the users created in the super method
+        res = super(PortalWizard, self).onchange_portal_id()
+
+        self.user_ids.write({
+            'invitation_config_id': self.invitation_config_id.id
+        })
+
+        return res
+
     @api.multi
     def action_apply(self):
         """
-        Generate communication for User invitation
+        return a recordset that contain a communication
+        with each user in the portal
         """
-        res = super(PortalWizard, self).action_apply()
-        comm_obj = self.env['partner.communication.job']
-        config = self.env.ref(
-            'partner_communication_switzerland.portal_welcome_config')
-        for user in self.mapped('user_ids.user_id'):
-            comm_obj.create({
-                'partner_id': user.partner_id.id,
-                'object_ids': user.id,
-                'config_id': config.id
-            })
-        return res
+        self.ensure_one()
```

Comment:
You removed the call to the super which here can remove some useful features. Can you add it back?

Reply:
No, because the parent didn't return the result of the action_apply method

```
    @api.multi
    def action_apply(self):
        self.ensure_one()
        self.user_ids.action_apply()
        return {'type': 'ir.actions.act_window_close'}
```
Override had to be done there

Reply:
The problem is that this code will never be executed :
https://github.com/compassionch/compassion-modules/blob/bac7e809eb6deabea16c080698f84122601a69d9/crm_compassion/wizards/portal_wizard.py#L21
You should call the super even if it doesn't return what we need and then return what you need after. By calling super method you also don't need to call by yourself `self.user_ids.action_apply()`

Reply:
ok

Just one question, is the fact that the method portal.wizard.user action_apply() will be call once in the super method then again just to get the result is a problem ?

Reply:
I will try with a additional field in portal.wizard.user to not call twice the method

Reply:
it's a success